core/src/main/java/org/infinispan/util/IntSet.java

```diff
@@ -0,0 +1,196 @@
+package org.infinispan.util;
+
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.PrimitiveIterator;
+import java.util.Set;
+
+/**
+ * Represent a set of integers (e.g. segments) as a {@code BitSet}.
+ *
+ * @author Dan Berindei
+ * @since 9.0
+ */
+public class IntSet implements Set<Integer> {
+   BitSet segments;
+
+   public IntSet() {
```

Comment:
I would remove this ctor to prevent accidental use with too big ranges - user should be aware of the max.